dbscripts/xml/upgrade.xml

```diff
@@ -13,6 +13,7 @@
 
 <install version="3.5.0.0">
 	<code function="checkPhpVersion" />
+	<code function="addAppKey" />
```

Comment:
This should be added as a migration, not a `<code ...>` addition in the `Upgrade` class.

Reply:
@asmecher I understand it should part of upgrade and need to moved to upgrade class but I am confused about the migration part . I does not modify/update any DB tables but update the config.inc.php to add a unique app key if the `app_key` variable defined/set in the `config.inc.php` file . thats why it's added as `<code ...>` to run. or I am misunderstanding something ?

Reply:
The `<code ...>` pattern leads to accumulating cruft in the `Upgrade` class, so we're using Migration classes instead for new code. Migrations are good for whatever, not just database schema adaptations!